common/str.cpp

```diff
@@ -313,6 +313,11 @@ String String::vformat(const char *fmt, va_list args) {
 	return output;
 }
 
+//static
+String String::generateUniqueDomain( String gameId, int& suffixN) {
```

Comment:
This does not belong here, please move it to `EngineManager`.

Also, code formatting is off